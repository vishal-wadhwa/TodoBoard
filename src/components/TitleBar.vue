<template>
  <v-toolbar app fixed clipped-left>
    <slot name='toolbar-icon'></slot>
    <slot name='toolbar-heading'>
      <v-toolbar-title class='headline text-uppercase'>
        <span>{{title}}</span>
      </v-toolbar-title>
    </slot>
    <v-spacer/>
    <slot name='toolbar-content-right'></slot>
  </v-toolbar>
</template>
<script>
export default {
  props: {
    title: String
  }
}
</script>
<style lang="stylus" scoped>
</style>
