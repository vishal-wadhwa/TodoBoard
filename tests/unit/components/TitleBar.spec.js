import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import TitleBar from '@/components/TitleBar'
import Vuetify from 'vuetify'
import Vue from 'vue'
Vue.use(Vuetify)

describe('TitleBar.vue', () => {
  // required to vuetify [warn] in test
  const app = document.createElement('div')
  app.setAttribute('data-app', true)
  document.body.appendChild(app)

  // suppresses vuetify internal TypeError, tests pass regardless
  console.log = jest.fn()

  let localVue = null
  const factory = (opts = {}, full = false) => {
    return full ? mount(TitleBar, opts) : shallowMount(TitleBar, opts)
  }
  beforeEach(() => {
    localVue = createLocalVue()
    // localVue.use(Vuetify)
  })

  const title = 'TITLE NAME'

  it('renders title from props', () => {
    const wrapper = factory({
      propsData: { title },
      localVue
    })
    expect(wrapper.text()).toMatch(title)
  })

  // it('does not show option buttons and their tooltips by default', () => {
  //   const wrapper = factory({
  //     localVue
  //   }, true)

  //   // only one button that is the trigger itself
  //   expect(wrapper.findAll('.v-speed-dial .v-btn').length).toBe(1)
  //   expect(wrapper.findAll('.v-tooltip').length).toBe(0)
  // })

  // it('shows option buttons with tooltips when trigger button is clicked', () => {
  //   const wrapper = factory({
  //     localVue
  //   }, true)

  //   const triggerBtn = wrapper.find('.v-speed-dial .v-btn')
  //   triggerBtn.trigger('click')
  //   expect(wrapper.findAll('.v-speed-dial .v-btn').length).toBe(3)
  //   expect(wrapper.findAll('.v-tooltip').length).toBe(2)
  // })

  // it('emits "tb:create-list" when its respective button is clicked', () => {
  //   const wrapper = factory({
  //     localVue
  //   }, true)

  //   const triggerBtn = wrapper.find('.v-speed-dial .v-btn')
  //   triggerBtn.trigger('click')

  //   const createListBtn = wrapper.findAll('.v-speed-dial .v-btn').at(1)
  //   createListBtn.trigger('click')

  //   const emitObj = wrapper.emitted('tb:create-list')
  //   expect(emitObj[0][0]).toBeInstanceOf(Event)
  // })

  // it('emits "tb:create-board" when its respective button is clicked', () => {
  //   const wrapper = factory({
  //     localVue
  //   }, true)

  //   const triggerBtn = wrapper.find('.v-speed-dial .v-btn')
  //   triggerBtn.trigger('click')

  //   const createListBtn = wrapper.findAll('.v-speed-dial .v-btn').at(2)
  //   createListBtn.trigger('click')

  //   const emitObj = wrapper.emitted('tb:create-board')
  //   expect(emitObj[0][0]).toBeInstanceOf(Event)
  // })

  it('should match snapshot', () => {
    const wrapper = factory({
      localVue,
      propsData: { title }
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('should match snapshot when trigger button is clicked', () => {
  //   const wrapper = factory({
  //     localVue,
  //     propsData: { title }
  //   }, true)

  //   wrapper.find('.v-speed-dial .v-btn').trigger('click')
  //   expect(wrapper.html()).toMatchSnapshot()
  // })
})
