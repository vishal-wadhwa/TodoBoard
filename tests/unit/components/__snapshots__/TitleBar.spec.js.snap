// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar.vue should match snapshot 1`] = `
<nav class="v-toolbar v-toolbar--clipped v-toolbar--fixed theme--light" style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);">
  <div class="v-toolbar__content" style="height: 64px;">
    <div class="v-toolbar__title headline text-uppercase"><span>TITLE NAME</span></div>
    <div class="spacer"></div>
  </div>
</nav>
`;
